docs/src/developing/clients/jsonrpc-api.md

```diff
@@ -3443,6 +3445,175 @@ Result:
+
+**This subscription is unstable and only available if the validator was started
+with the `--rpc-pubsub-enable-block-subscription` flag. The format of this
+subscription may change in the future**
+
+Subscribe to receive notification anytime a new block is Confirmed or Finalized.
+
+#### Parameters:
+
+- `filter: <string>|<object>` - filter criteria for the logs to receive results by account type; currently supported:
+  - "all" - include all transactions in block
+  - `{ "mentionsAccountOrProgram": <string> }` - filter out transactions that don't mention the provided public key (as base-58 encoded string), if no mentions for a given block, then no notification will be published
+- `<object>` - (optional) Configuration object containing the following optional fields:
+  - (optional) [Commitment](jsonrpc-api.md#configuring-state-commitment)
+  - (optional) `encoding: <string>` - encoding for Account data, either "base58" (*slow*), "base64", "base64+zstd" or "jsonParsed".
+    "jsonParsed" encoding attempts to use program-specific state parsers to return more human-readable and explicit account state data. If "jsonParsed" is requested but a parser cannot be found, the field falls back to base64 encoding, detectable when the `data` field is type `<string>`. Default is "base64".
+  - (optional) `transactionDetails: <string>` - level of transaction detail to return, either "full", "signatures", or "none". If parameter not provided, the default detail level is "full".
+  - (optional) `showRewards: bool` - whether to populate the `rewards` array. If parameter not provided, the default includes rewards.
+
+#### Results:
+
+- `integer` - subscription id \(needed to unsubscribe\)
+
+#### Example:
+
+Request:
+```json
+{"jsonrpc": "2.0", "id": "1", "method": "blockSubscribe", "params": ["all"]}
+```
+```json
+{
+  "jsonrpc": "2.0",
+  "id": "1",
+  "method": "blockSubscribe",
+  "params": [
+    {"mentionsAccountOrProgram": "3UVYmECPPMZSCqWKfENfuoTv51fTDTWicX9xmBD2euKe"},
+    {
+      "commitment": "confirmed",
+      "encoding": "base64",
+      "showRewards": true,
+      "transactionDetails": "full"
+    }
+  ]
+}
+```
+
+Result:
+```json
+{"jsonrpc": "2.0","result": 0,"id": 1}
+```
+
+#### Notification Format:
+
+The notification will be an object with the following fields:
+
+- `signature: <string>` - The transaction signature base58 encoded.
+- `err: <object | null>` - Error if something went wrong publishing the notification otherwise null.
+- `logs: <array | null>` - Array of log messages the transaction instructions output during execution, null if simulation failed before the transaction was able to execute (for example due to an invalid blockhash or signature verification failure)
+- `block: <object | null>` - A block object as seen in the [getBlock](jsonrpc-api.md#getblock) RPC HTTP method.
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "blockNotification",
+  "params": {
+    "result": {
+      "context": {
+        "slot": 5208469
```

Comment:
This will likely be within a few slots of the block being notified, won't it? Let's make this something realistic.